linxiangxi/client/src/client.cpp

```diff
@@ -0,0 +1,68 @@
+#include <ros/ros.h>
+#include <csignal>
+#include <client/client_message.h>
+#include <client/client_login.h>
+
+#define PUBLISHER_SIZE 10
+
+#define LOGIN 0
+#define LOGOUT 1
+#define NUL 2
+#define SUCCESS 3
+#define FAILED 4
+
+client::client_login * login_service_ptr = nullptr;
+ros::ServiceClient * client_login_ptr = nullptr;
+
+void signal_handler(int signal){
+    ROS_INFO("logout");
+    (*login_service_ptr).request.require_code = LOGOUT;
+    (*login_service_ptr).response.answer_code = NUL;
+    while(true){
+        if((*client_login_ptr).call(*login_service_ptr)){
+            if((*login_service_ptr).response.answer_code == SUCCESS){
+                break;
+            }
+        }
+    }
+    ros::shutdown();
+}
+
+int main(int argc, char ** argv){
+    ros::init(argc, argv, "client", ros::init_options::AnonymousName);
+    ros::NodeHandle client_node_handle;
+    ros::Publisher client_pub = client_node_handle.advertise<client::client_message>(ros::this_node::getName(), PUBLISHER_SIZE);
+    ros::ServiceClient client_login = client_node_handle.serviceClient<client::client_login>("client_login");
```

Comment:
**[cpplint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br> Lines should be <= 80 characters long  [whitespace/line_length] [2]